app/assets/javascripts/repository_columns/columns_initializers/list_column_type.js

```diff
@@ -0,0 +1,137 @@
+/* global GLOBAL_CONSTANTS  I18n */
+/* eslint-disable no-unused-vars */
+var RepositoryListColumnType = (function() {
+  var manageModal = '#manage-repository-column';
+
+  function onlyUnique(value, index, self) {
+    return self.indexOf(value) === index;
+  }
+
+  function textToItems(text, delimiter) {
+    var res = [];
+    var usedDelimiter = '';
+    var definedDelimiters = {
+      return: '\n',
+      comma: ',',
+      semicolon: ';',
+      space: ' '
+    };
+
+    var delimiters = [];
+    if (delimiter === 'auto') {
+      delimiters = ['\n', ',', ';', ' '];
+    } else {
+      delimiters.push(definedDelimiters[delimiter]);
+    }
+
+    $.each(delimiters, (index, currentDelimiter) => {
+      res = text.trim().split(currentDelimiter);
+      usedDelimiter = Object
+        .keys(definedDelimiters)
+        .find(key => definedDelimiters[key] === currentDelimiter);
+
+      if (res.length > 1) {
+        return false;
+      }
+      return true;
+    });
+
+    res = res.filter(Boolean).filter(onlyUnique);
+
+    $.each(res, (index, option) => {
+      res[index] = option.slice(0, GLOBAL_CONSTANTS.NAME_MAX_LENGTH);
+    });
+
+    $('select#delimiter').attr('data-used-delimiter', usedDelimiter);
+    return res;
+  }
+
+  function pluralizeWord(count, noun, suffix = 's') {
+    return `${noun}${count !== 1 ? suffix : ''}`;
+  }
+
+
+  function drawDropdownPreview(items) {
+    var $manageModal = $(manageModal);
+    var $dropdownPreview = $manageModal.find('.dropdown-preview select');
+    $('option', $dropdownPreview).remove();
+    $.each(items, function(i, item) {
+      $dropdownPreview.append($('<option>', {
+        value: item,
+        text: item
+      }));
+    });
+
+    if (items.length === 0) {
+      $dropdownPreview.append($('<option>', {
+        value: '',
+        text: I18n.t('libraries.manange_modal_column.list_type.dropdown_item_select_option')
+      }));
+    }
+  }
+
+  function refreshCounter(number) {
+    var $manageModal = $(manageModal);
+    $manageModal.find('.list-items-count').html(number);
+    $manageModal.find('.list-items-count').attr('data-count', number);
+
+    if (number >= GLOBAL_CONSTANTS.REPOSITORY_LIST_ITEMS_PER_COLUMN) {
+      $manageModal.find('.limit-counter-container').addClass('error-to-many-items');
+      $manageModal.find('button[data-action="save"]').addClass('disabled');
+    } else {
+      $manageModal.find('.limit-counter-container').removeClass('error-to-many-items');
+      $manageModal.find('button[data-action="save"]').removeClass('disabled');
```

Comment:
create different variables to counter-container and data-action. And better add some class to button, don't use data attribute.